### PR TITLE
Fix GH-248

### LINKED ISSUE #248: ValueError: Line in unexpected format
Hi, 

This code raise exception
```python
from eli5 import show_prediction
show_prediction(gbtree.get_booster(), df[ds_cols].iloc[1])
``` 

Traceback
```
---------------------------------------------------------------------------
ValueError                                Traceback (most recent call last)
<ipython-input-84-8f8dbfe0c6ec> in <module>()
----> 1 show_prediction(gbtree.get_booster(), df[ds_cols].iloc[0])

/usr/local/lib/python3.6/site-packages/eli5/ipython.py in show_prediction(estimator, doc, **kwargs)
    268     """
    269     format_kwargs, explain_kwargs = _split_kwargs(kwargs)
--> 270     expl = explain_prediction(estimator, doc, **explain_kwargs)
    271     html = format_as_html(expl, **format_kwargs)
    272     return HTML(html)

/usr/local/lib/python3.6/site-packages/singledispatch.py in wrapper(*args, **kw)
    208 
    209     def wrapper(*args, **kw):
--> 210         return dispatch(args[0].__class__)(*args, **kw)
    211 
    212     registry[object] = func

/usr/local/lib/python3.6/site-packages/eli5/xgboost.py in explain_prediction_xgboost(xgb, doc, vec, top, top_targets, target_names, targets, feature_names, feature_re, feature_filter, vectorized, is_regression, missing)
    194 
    195     scores_weights = _prediction_feature_weights(
--> 196         booster, dmatrix, n_targets, feature_names, xgb_feature_names)
    197 
    198     x = get_X0(add_intercept(X))

/usr/local/lib/python3.6/site-packages/eli5/xgboost.py in _prediction_feature_weights(booster, dmatrix, n_targets, feature_names, xgb_feature_names)
    258             ) for target_idx in range(n_targets)]
    259     else:
--> 260         scores_weights = [target_feature_weights(leaf_ids, tree_dumps)]
    261     return scores_weights
    262 

/usr/local/lib/python3.6/site-packages/eli5/xgboost.py in _target_feature_weights(leaf_ids, tree_dumps, feature_names, xgb_feature_names)
    271     score = 0
    272     for text_dump, leaf_id in zip(tree_dumps, leaf_ids):
--> 273         leaf = _indexed_leafs(_parse_tree_dump(text_dump))[leaf_id]
    274         score += leaf['leaf']
    275         path = [leaf]

/usr/local/lib/python3.6/site-packages/eli5/xgboost.py in _parse_tree_dump(text_dump)
    339     for line in text_dump.split('\n'):
    340         if line:
--> 341             depth, node = _parse_dump_line(line)
    342             if depth == 0:
    343                 assert not stack

/usr/local/lib/python3.6/site-packages/eli5/xgboost.py in _parse_dump_line(line)
    384             'cover': float(cover),
    385         }
--> 386     raise ValueError('Line in unexpected format: {}'.format(line))
    387 
    388 

ValueError: Line in unexpected format: 0:[path_type_/items/black_rook] yes=2,no=1,gain=0.443101,cover=371490
```

I think problem is in -> https://github.com/TeamHG-Memex/eli5/blob/eeb37353b2309357468c57349df75b41604e8353/eli5/xgboost.py#L359
My DataFrame have `boolean` values only, so there is no symbol `<` in nodes.

### PR DESCRIPTION
**Pull request by sharthZ**
I think, this will fix #248 
Can't run tests, because in my version of xgboost `xgb.booster()` must be replaced by `xgb.get_booster()`

### CODE REVIEW

**Comment by gabrielelanaro** [comment]
Any chance this pull request will be merged? This is quite an important bug

**Comment by lopuhin** [comment]
Sorry for slow feedback @sharthZ and @gabrielelanaro , I think there are two items preventing the merge:
- it would be nice to add a test to this PR that checks that this does fix the issue with boolean attributes
- unrelated to this PR: unfortunately, eli5 does not work with xgboost 0.7 at the moment which causes tests to fail, so in order to run them we need to either pin xgboost to <0.7 temporary in tests, or to fix the issues with xgboost 0.7
